lib/matplotlib/colors.py

```diff
@@ -958,6 +958,677 @@ def scaled(self):
+        refpoints_data = np.array([center_data])
+
+        flist = [(lambda x:(-fneg(-x + 1) + 1)), fpos]
+        finvlist = [(lambda x:(-fneginv(-x + 1) + 1)), fposinv]
+
+        (super(MirrorPiecewiseNorm, self)
+         .__init__(flist=flist,
+                   finvlist=finvlist,
+                   refpoints_cm=refpoints_cm,
+                   refpoints_data=refpoints_data,
+                   **normalize_kw))
+
+
+class MirrorRootNorm(MirrorPiecewiseNorm):
+    """
+    Root normalization for positive and negative data.
+
+    Data above `center_data` will be normalized with a root of the order
+    `orderpos` and data below it with a symmetric root of order `orderg neg`.
+    If only `orderpos` function is given, the normalization will be completely
+    mirrored.
+    """
+
+    def __init__(self, orderpos=2, orderneg=None,
+                 center_cm=0.5, center_data=0.0,
+                 **normalize_kw):
+        """
+        Parameters
+        ----------
+        orderpos : float or int, optional
+            Degree of the root to be used for normalization of values
+            above `center_data` using `MirrorPiecewiseNorm`. Default 2.
+        fneg, fneginv : callable or string, optional
+            Degree of the root to be used for normalization of values
+            below `center_data` using `MirrorPiecewiseNorm`. Default
+            `orderpos`.
+        center_data : float, optional
+            Value in the data that will separate range. Must be
+            in the (vmin, vmax) range.
+            Default 0.0.
+        center_cm : float, optional
+            Normalized value that will correspond do `center_data`. Must be
+            in the (0.0, 1.0) range.
+            Default 0.5.
+        normalize_kw : dict, optional
+            Dict with keywords (`vmin`,`vmax`,`clip`) passed
+            to `matplotlib.colors.Normalize`.
+
+        Examples
+        --------
+        Obtaining a symmetric amplification of the features around 0:
+
+        >>> import matplotlib.colors as colors
+        >>> norm = mcolors.MirrorRootNorm(orderpos=2)
+
+        Obtaining an asymmetric amplification of the features around 0.6:
+
+        >>> import matplotlib.colors as colors
+        >>> norm = mcolors.MirrorRootNorm(orderpos=3,
+        >>>                               orderneg=4,
+        >>>                               center_data=0.6,
+        >>>                               center_cm=0.3)
+
+        """
+
+        if orderneg is None:
+            orderneg = orderpos
+        (super(MirrorRootNorm, self)
+         .__init__(fneg=(lambda x: x**(1. / orderneg)),
+                   fneginv=(lambda x: x**(orderneg)),
+                   fpos=(lambda x: x ** (1. / orderpos)),
+                   fposinv=(lambda x: x**(orderpos)),
+                   center_cm=center_cm,
+                   center_data=center_data,
+                   **normalize_kw))
+
+
+class RootNorm(FuncNorm):
+    """
```

Comment:
At the least I'm in favor of documenting that in docstring,  but I'd also lean towards taking it out if it's supported fairly cleanly in the other functions.